### PR TITLE
jhipster-terraform-azure

### PR DESCRIPTION
**Pull request by indiepopart**
This blog post has the following features:

- [x] A GitHub Repository with a polished README
- [ ] A GitHub Repository under the github.com/oktadev account
- [ ] A title that's approved by Dev Advocacy
- [ ] A URL approved by Dev Advocacy
- [x] The content has been run through Grammarly (https://www.grammarly.com/)
- [x] Rendered locally and confirmed that no Markdown typos exist
- [ ] Images are compressed appropriately
- [ ] Social image previews well on Twitter and LinkedIn
- [ ] Tech review request (developer advocate or domain expert)
- [ ] Review request for editorial/grammar/clarity (developer advocate or Colton)

### CODE REVIEW

File: _source/_posts/2024-07-05-jhipster-terraform-azure.md [hunk 1]
```diff
@@ -0,0 +1,1857 @@
...
+> - [Auth0 CLI 1.4.0](https://github.com/auth0/auth0-cli#installation)
+> - [Docker 26.1.2](https://docs.docker.com/desktop/)
+> - [Azure account](https://azure.microsoft.com/en-us/pricing/purchase-options/pay-as-you-go)
+> - [Azure CLI 2.60.0](https://learn.microsoft.com/en-us/cli/azure/install-azure-cli)
+> - [JHipster 8.4.0](https://www.jhipster.tech/)
+> - [kubectl 1.30.1](https://kubernetes.io/docs/tasks/tools/#kubectl)
+> - [Terraform 1.8.3](https://developer.hashicorp.com/terraform/install)
+> - [jq 1.6](https://jqlang.github.io/jq/download/)
+
+{% include toc.md %}
```
Comment (deepu105) [inline _source/_posts/2024-07-05-jhipster-terraform-azure.md]:
```suggestion

```

File: _source/_posts/2024-07-05-jhipster-terraform-azure.md [hunk 2]
```diff
@@ -0,0 +1,1857 @@
...
+
+## Build a microservices architecture with JHipster
+
+Create a Java microservices architecture using [JHipster](https://www.jhipster.tech/), Spring Boot, and Consul. JHipster is an excellent tool for generating a microservice stack with Spring Boot, Angular/React/Vue.js, and other modern frameworks. For deploying the application of this demo, then you can either generate it using JHipster JDL or clone the sample repository from GitHub. Here is how you can build your microservices stack using JHipster:
```
Comment (deepu105) [inline _source/_posts/2024-07-05-jhipster-terraform-azure.md]:
```suggestion
Create a Java microservices architecture using [JHipster](https://www.jhipster.tech/), Spring Boot, and Consul. JHipster is an excellent tool for generating a microservice stack with Spring Boot, Angular/React/Vue.js, and other modern frameworks. For deploying the application of this demo, you can either generate it using JHipster JDL or clone the sample repository from GitHub. Here is how you can build your microservices stack using JHipster:
```

File: _source/_posts/2024-07-05-jhipster-terraform-azure.md [hunk 3]
```diff
@@ -0,0 +1,1857 @@
...
+
+**Option 1**: Generate the architecture with JHipster Domain Language (JDL)
+
+```shell
+mkdir jhipster-microservice-stack
+cd jhipster-microservice-stack
+# download the JDL file.
+jhipster download https://raw.githubusercontent.com/indiepopart/jhipster-terraform-azure/main/apps.jdl
```
Comment (deepu105) [inline _source/_posts/2024-07-05-jhipster-terraform-azure.md]: please transfer the repo to oktadev and update the rfs to it

File: _source/_posts/2024-07-05-jhipster-terraform-azure.md [hunk 4]
```diff
@@ -0,0 +1,1857 @@
...
+touch main.tf
+touch outputs.tf
+touch variables.tf
+```
+
+Edit `main.tf` and add the following content:
+
+```terraform
+# terraform/modules/hub_network/main.tf
+locals {
```
Comment (deepu105) [inline _source/_posts/2024-07-05-jhipster-terraform-azure.md:144]: @indiepopart do you thibnk we can make this script shorter by removing fields that are optional and has a default value? This is too long for a blog IMO

File: _source/_posts/2024-07-05-jhipster-terraform-azure.md [hunk 5]
```diff
@@ -0,0 +1,1857 @@
...
+touch main.tf
+touch outputs.tf
+touch variables.tf
+```
+
+Edit `main.tf` and add the following content:
+
+```terraform
+# terraform/modules/spoke_network/main.tf
+locals {
```
Comment (deepu105) [inline _source/_posts/2024-07-05-jhipster-terraform-azure.md:463]: Same here can we exlcude any field that is just setting a default value?

File: _source/_posts/2024-07-05-jhipster-terraform-azure.md [hunk 6]
```diff
@@ -0,0 +1,1857 @@
...
+    backend_http_settings_name = local.http_setting_name
+  }
+}
+```
+The configuration above will create a Spoke Network, the network peerings the hub and spoke networks, and the Azure application gateway hosted in a dedicated subnet.
+
+Edit `outputs.tf` and add the following content:
+
+```terraform
+# terraform/modules/spoke_network/outputs.tf
```
Comment (deepu105) [inline _source/_posts/2024-07-05-jhipster-terraform-azure.md:536]: Lets also remove outputs that are not useful for any further usecase

**Review by deepu105 — CHANGES_REQUESTED** [review]
The blog post lookks fine but the TF scripts are too long, we should exclude anything that is unnecessary abd default to try make this shorter

File: _source/_posts/2024-07-05-jhipster-terraform-azure.md (hunk 3 above)
Comment (indiepopart) [reply _source/_posts/2024-07-05-jhipster-terraform-azure.md]: I don't have privileges for creating repos in oktadev. I transferred to you.

File: _source/_posts/2024-07-05-jhipster-terraform-azure.md (hunk 4 above)
Comment (indiepopart) [reply _source/_posts/2024-07-05-jhipster-terraform-azure.md:144]: Did a split and removed some optionals.

File: _source/_posts/2024-07-05-jhipster-terraform-azure.md (hunk 5 above)
Comment (indiepopart) [reply _source/_posts/2024-07-05-jhipster-terraform-azure.md:463]: Did a split and removed some optionals.

File: _source/_posts/2024-07-05-jhipster-terraform-azure.md (hunk 6 above)
Comment (indiepopart) [reply _source/_posts/2024-07-05-jhipster-terraform-azure.md:536]: Removed unused outputs.

**Comment by indiepopart** [comment]
@deepu105 Removed optionals and defaults, and split the scripts. Added some comments about Azure. Back to you!

File: _source/_posts/2024-07-05-jhipster-terraform-azure.md [hunk 7]
```diff
@@ -0,0 +1,1699 @@
...
+az account list
+```
+
+> **IMPORTANT NOTE**
+> For this demo, the selected VM size is __Standard_B2s_v2__, and the deployed architecture requires a **minimum node count of 4**. The architecture will not run under the Azure free account, so please don't forget to delete the architecture after the test to avoid unwanted costs.
+
+Verify you have the available cores quota for the minimum node count of 4 (8 cores):
+
+```shell
+az quota show --resource-name standardBsv2Family --scope /subscriptions/<account-id>/providers/Microsoft.Compute/locations/eastus2
```
Comment (deepu105) [inline _source/_posts/2024-07-05-jhipster-terraform-azure.md:1047]:
```suggestion
# `<account-id>` is the `id` field from the az account list output
az quota show --resource-name standardBsv2Family --scope /subscriptions/<account-id>/providers/Microsoft.Compute/locations/eastus2
```

**Review by deepu105 — APPROVED** [review]

**Comment by deepu105** [comment]
closing as its moved to auth0 blog

**Pull request closed without merging**